src/types.ts

```diff
@@ -45,6 +45,10 @@ type UnpackAll<List, output extends unknown[] = []> = List extends [
   ? UnpackAll<rest, [...output, first]>
   : output
 
+type UnpackDFObject<Obj extends Record<string, DomainFunction>> =
+  | { [K in keyof Obj]: UnpackData<Obj[K]> }
+  | never
```

Comment:
As in the #65 this `| never` will prettify the resulting type of the composition.